core/interval/int.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2024 Canonical Ltd.
```

Comment:
I don't want to be a killjoy, but it looks like there are some official go libraries to do this kind of thing https://pkg.go.dev/golang.org/x/tools/container/intsets

Reply:
intsets seems to do something else, it's a data structure representing sparse integer series.

I guess you could argue a a port range is a sparse integer list. We have 65,000 odd closed ports initially, then we open some like 443 or what not.

However, it's perfectly reasonable someone may (in development) want to open ports 0 - 10,000. This would be very inefficient for a sparse set. 

I have had a cursory look if there are any suitable data structures in the market, but wasn't really able to find any. I'll look deeper now

Reply:
I have not managed to find anything appropriate so far.

I get the impression that the philosophy of Golang tends to shy away from providing these kinds of utilities, encouraging engineers to think more about specific implementations each time. Hence why you have no build in slice filter, map, etc. functionality

Agree or disagree, that tends to be how it is, resulting in much boiler plate like this